cirq-core/cirq/protocols/control_key_protocol.py

```diff
@@ -34,7 +35,7 @@ class SupportsControlKey(Protocol):
     """
 
     @doc_private
-    def _control_keys_(self) -> Iterable['cirq.MeasurementKey']:
+    def _control_keys_(self) -> FrozenSet['cirq.MeasurementKey']:
```

Comment:
This is more complicated than the `AbstractSet` / `FrozenSet` case - we need to ensure that internal code doesn't take advantage of set-like behavior from this until after a release.

Reply:
Yeah, this is handled by lines 60-65 in this file.

Reply:
Is this what you meant in your top-level comment? Old external magic will get this warning. Nothing specifically relies on this frozenness, at least not that I know of.